### ISSUE TITLE
mpl drawer: unitary gate looks messy for 1 qubit

### ISSUE THREAD

**Issue by ajavadia** [Qiskit/qiskit-terra#2233]
### What is the expected enhancement?
If I add unitary gates to the circuit, they show up fine in the `mpl` drawer except for the 1-qubit case. The single-qubit case should also not print the full matrix, and just do a box labeled "unitary".

```python
from qiskit import QuantumRegister, QuantumCircuit
from qiskit.quantum_info.random import random_unitary
qr = QuantumRegister(4)
qc = QuantumCircuit(qr)
U1 = random_unitary(2)
U2 = random_unitary(4)
U3 = random_unitary(8)
U4 = random_unitary(16)
qc.append(U1, [qr[0]])
qc.append(U2, [qr[0], qr[1]])
qc.append(U3, [qr[0], qr[1], qr[2]])
qc.append(U4, [qr[0], qr[1], qr[2], qr[3]])
qc.draw(output='mpl')
```
![image](https://user-images.githubusercontent.com/8622381/56902007-64fd2c00-6a67-11e9-81f5-79c985cb4dab.png)

**Comment by maddy-tod** [comment]
It looks like the box isn't in the correct place for the 4-qubit case? Potentially a separate issue

**Comment by ajavadia** [comment]
It's a bit of an offset, probably due to that 0 label at the top. Not a big deal but yes would be nice to fix that too.

**Comment by mtreinish** [comment]
> It looks like the box isn't in the correct place for the 4-qubit case? Potentially a separate issue

Yeah, they look like 2 separate issues. It looks like the dynamic sizing for the 4 qubit case could be off slightly. (although my eyes may be playing ticks on me as I think I see the spacing between qubit 0 and 1 as slightly larger) The label issue is probably caused by the param fix in #2169 treating the matrix as a string parameter. We'll probably just need to add real type checking there.
